src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/ReconciliationReportRepositoryTest.java

```diff
@@ -117,4 +135,76 @@ void should_return_empty_optional_if_report_does_not_exist() {
         // then
         assertThat(report).isEmpty();
     }
+
+    @Test
+    void should_not_find_anything_when_db_is_empty() {
+        // when
+        Optional<ReconciliationContent> report = reportRepo.getLatestReconciliationReport(now(), ACCOUNT);
+
+        // then
+        assertThat(report).isEmpty();
+    }
+
+    @Test
+    void should_not_find_anything_when_conditions_do_not_match() {
+        // given
+        saveNewReportsAndGetLastId("{}");
+
+        // when
+        Optional<ReconciliationContent> report = reportRepo.getLatestReconciliationReport(now().minusDays(1), ACCOUNT);
+
+        // then
+        assertThat(report).isEmpty();
+    }
+
+    @Test
+    void should_find_a_report_when_conditions_are_met() {
+        // given
+        var expectedReportContent = "{}";
+        var id = saveNewReportsAndGetLastId(expectedReportContent);
+
+        // when
+        Optional<ReconciliationContent> report = reportRepo.getLatestReconciliationReport(now(), ACCOUNT);
+
+        // then
+        assertThat(report)
+            .isNotEmpty()
+            .get()
+            .usingRecursiveComparison()
+            .isEqualTo(new ReconciliationContent(id, expectedReportContent, VERSION));
+    }
+
+    @Test
+    void should_find_only_latest_report_when_conditions_are_met() {
+        // given
+        var expectedReportContent = "[]";
+        var id = saveNewReportsAndGetLastId("{}", expectedReportContent);
+
+        // when
+        Optional<ReconciliationContent> report = reportRepo.getLatestReconciliationReport(now(), ACCOUNT);
+
+        // then
+        assertThat(report)
+            .isNotEmpty()
+            .get()
+            .usingRecursiveComparison()
+            .isEqualTo(new ReconciliationContent(id, expectedReportContent, VERSION));
+    }
+
+    // Only last ID/report matters
+    private UUID saveNewReportsAndGetLastId(String... reportContents) {
+        UUID lastId = null;
+
+        try {
+            for (String contents : reportContents) {
+                var statementId = statementRepo.save(NEW_STATEMENT);
+                var report = new NewReconciliationReport(statementId, ACCOUNT, contents, VERSION);
+                lastId = reportRepo.save(report);
+            }
+        } catch (SQLException exception) {
+            throw new RuntimeException("Could not save reports for testing", exception);
+        }
```

Comment:
or you can just have standard `throws Exception` at the end of test signature

Reply:
like most of the tests that existed here before.

Reply:
don't want that throws signature. especially generic exception. especially in tests